Remove DistExecutor calls in PistonEventTest

sciwhiz12 opened the pull request:
The use of `DistExecutor#safeCallWhenOn` in the `PistonEventTest` causes a crash when a piston extends or retracts in the test mod runs, because `safeCallWhenOn` is passed a double lambda which is not a safe referent. (A safe referent is a lambda that returns a method reference.)

~~This PR fixes those crashes by changing to the unsafe variant, `unsafeCallWhenOn`, which does not check if the passed in lambda is a safe referent. This removes a long-time annoyance from the test sourceset runs, because the crash often occurs when testing other PRs (with a mechanism that uses a piston).~~

This PR fixes those crashes by removing the references to `DistExecutor` and the client local player and replacing message sending calls to broadcast to all players on the server. In addition, a constant is added to control whether these messages are actually sent, to avoid spamming the chat in case a test world uses many pistons. By default, the messages are disabled.

This PR also cleans up the warnings and formatting of the test mod.

sciwhiz12 committed:
Fix DistExecutor calls in PistonEventTest

The use of lambdas leads to a crash due to not being a safe referent.
Switching to the unsafe variant, unsafeCallWhenOn, fixes the crash.

Matyrobbrt reviewed (approved):
Got to love `DistExecutor`...

tmvkrpxl0 commented:
Speaking of which, should we get rid of DistExecutor#validateSafeReferent? iirc it doesn't work at all on java9+

LexManos commented:
No, the point is that this crashes when things should crash. or atleast where its detectable.
Proper solution would be to move to method references or whatever doesnt load client classes on the server. Not to use unsafe. But this is a test mod so doesnt really matter.

sciwhiz12 committed:
Broadcast messages to all players instead of local player

This removes the need to check the local player instance in Minecraft,
and therefore the uses of DistExecutor.

sciwhiz12 committed:
Reorganize imports

sciwhiz12 committed:
Cleanup warnings and formatting

sciwhiz12 committed:
Add constant to control PistonEventTest logging

This avoids spamming the chat and logs if a test world happens to use
many pistons.

sciwhiz12 commented:
Removed the references to `DistExecutor` and the client local player, and changed how messages are sent so they are broadcast to all players instead. Please see the updated PR description for more details.
